### DISCUSSION TITLE
Windows Environment variables equivalent for AWS_ENV="SANDBOX"

### DISCUSSION THREAD

**Discussion by anova-git** [saleweaver/python-amazon-sp-api#1367] [FAQ] [answered]
I am running on a Windows 10 and export environment variables are not available. For credentials, I have the option to have them in the code or have a config file on the %APPDATA% folder (credentials.yml file). I have used both and they work fine but I cannot find a way to use other environment variables like AWS_ENV="SANDBOX" or SP_API_DEFAULT_MARKETPLACE='US'. 

An approach similar to the credentials config file would be nice but I am not sure it is available.

I have tried git-bash, bash.exe with "export" or command prompt with SET AWS_ENV="SANDBOX", but so far I can't access the Sandbox.

Any suggestions?

Environment:
Windows 10
VSCode
Python 3.12.2

on cmd prompt:
```
C:\Users\<.. user..> echo %AWS_ENV%
"SANDBOX"
```

**Comment by anova-git** [comment, marked as answer]
Although the AWS_ENV variable can be modified (and displayed) dynamically on Windows using bash, git-bash, com prompt command SET or Python os.environ['AWS_ENV'] = 'SANDBOX', a Window re-boot is needed before it can be recognized. In all the options above, the setting will be lost after the reboot, and the only way to have it permanent is to set it to the Windows Environment variables using the Windows  UI.

Reboot is needed to make the variable available to the system (also when switching back to PRODUCTION or just deleting the variable.

The SP-API Sandbox environment became available but unfortunately, it looks like there is nothing there. All reports requested failed and nothing I tried worked on the sandbox although they are fine on the production environment. There are threads here on the subject, but it is outside of the scope of the original discussion.

**Reply by saleweaver** [reply]
There are all endpoints available in the sandbox, but all of them return responses only for a preset request. The requests are layered out in the schema json files, available for each endpoint. 

The sandbox is extremely useless, I wouldn't spend time bothering with it. Every ChatGPT generated mock is better

**Reply by anova-git** [reply]
Thanks for sharing this information. I wasn't sure whether to pursue using the Sandbox. Now I know I shouldn't
